MessageHandler.js

```diff
@@ -25,14 +27,24 @@ class MessageHandler{
             return;
         }
 
-        // we only accept messages that start with "$"
-        if(!message || message.length < 1 || message[0] !== "$"){
+        // If this message starts with the comment prefix (and one is set)
+        // ignore this message no matter what
+        if (this.commentPrefix.length > 0 && message.startsWith(this.commentPrefix)){
```

Comment:
Why are you checking `length` here? we probably want `commentPrefix` to be `null` or `undefined` when we don't want to specify it.